soyjs/exec.go

```diff
@@ -510,44 +511,119 @@ func (s *state) visitSwitch(node *ast.SwitchNode) {
 	s.jsln("}")
 }
 
-// TODO: {msg} node translation is unimplemented
 func (s *state) visitMsg(node *ast.MsgNode) {
-	var pluralNode, ok = node.Body.Children()[0].(*ast.MsgPluralNode)
-	if !ok {
-		s.visitMsgBody(node)
+	// If no bundle was provided, walk the message sub-nodes.
+	if s.options.Messages == nil {
+		s.visitMsgNode(node)
 		return
 	}
 
-	s.jsln("switch (", pluralNode.Value, ") {")
+	// Look up the message in the bundle.
+	var msg = s.options.Messages.Message(node.ID)
+	if msg == nil {
+		s.visitMsgNode(node)
+		return
+	}
+
+	// Render each part.
+	s.evalMsgParts(node, msg.Parts)
+}
+
+func (s *state) evalMsgParts(msgNode *ast.MsgNode, parts []soymsg.Part) {
+	for _, part := range parts {
+		switch part := part.(type) {
+
+		case soymsg.RawTextPart:
+			s.writeRawText([]byte(part.Text))
+
+		case soymsg.PlaceholderPart:
+			// Find the node corresponding to the placeholder, and walk it.
+			var phnode = msgNode.Placeholder(part.Name)
+			if phnode == nil {
+				s.errorf("failed to find placeholder %q in %q",
+					part.Name, soymsg.PlaceholderString(msgNode))
+			}
+			s.walk(phnode.Body)
+
+		case soymsg.PluralPart:
+			// Find the corresponding node for this part.
+			child := s.findPluralNode(msgNode, part.VarName)
+			s.jsln("switch (", child.Value, ") {")
+			s.indentLevels++
+
+			for i, pluralCase := range child.Cases {
+				casePart := part.Cases[i]
+				s.jsln("case ", pluralCase.Value, ":")
+				s.indentLevels++
+				s.evalMsgParts(msgNode, casePart.Parts)
+				s.jsln("break;")
+				s.indentLevels--
+			}
+			{
+				s.jsln("default:")
+				s.indentLevels++
+
+				// The last pluralcase part should be the default case.
+				casePart := part.Cases[len(part.Cases)-1]
+
+				// TODO: I don't think this is right.  We probably need to use the message bundle's
+				// PluralCase method to somehow add more case statements here, but I'm not sure.
+				if len(part.Cases) > len(child.Cases) {
```

Comment:
Unfortunately, you need a JS version of all of the plural methods, and then you would use the plural selector calculated by the language specific function as an input the switch. I don't think that Closure Library's is fully compatible with PO format, but it would be something like this: 
https://github.com/google/closure-library/blob/master/closure/goog/i18n/pluralrules.js

I think it's basically porting these:
https://github.com/robfig/gettext/blob/master/po/plural.go

In theory, it should be taking the formula from what's specified in .po files, but I don't think that's necessary or needed